spikes/bullets/BulletManager.cpp

```diff
@@ -0,0 +1,51 @@
+#include "stdafx.h"
+#include "BulletManager.h"
+
+
+BulletManager * BulletManager::instance = 0;
+
+void BulletManager::addBullet(BulletType bullet, Vector2D position, Vector2D direction)
+{
+	Bullet myBullet;
+
+	//use this switch statement to pass in different spritesheets/other vars to different bullets depending on type
+	switch (bullet)
+	{
+	case BulletType::SQUARE:
+
+		//note that i pass the bullet type in here, this will be removed asap
+		BulletManager::getInstance()->bulletList.push_back(new Bullet(bullet, position, direction, 5));
```

Comment:
Well it will be dependant on the bullet type, but yes the speed of the
bullet coming from the gun will be determined by the type of gun you have,
these will obviously be implemented when we have multiple weapons to test
with
On 19 Oct 2015 11:19, "ShaneCallanan" <notifications@github.com> wrote:

> In spikes/bullets/BulletManager.cpp
> <https://github.com/gdxn96/swarm-wars/pull/4#discussion_r42355488>:
>
> > +#include "BulletManager.h"
> > +
> > +
> > +BulletManager * BulletManager::instance = 0;
> > +
> > +void BulletManager::addBullet(BulletType bullet, Vector2D position, Vector2D direction)
> > +{
> > +	Bullet myBullet;
> > +
> > +	//use this switch statement to pass in different spritesheets/other vars to different bullets depending on type
> > +	switch (bullet)
> > +	{
> > +	case BulletType::SQUARE:
> > +
> > +		//note that i pass the bullet type in here, this will be removed asap
> > +		BulletManager::getInstance()->bulletList.push_back(new Bullet(bullet, position, direction, 5));
>
> Make 'speed' member of Weapon?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdxn96/swarm-wars/pull/4/files#r42355488>.
>